prophyc/__init__.py

```diff
@@ -31,41 +36,61 @@ def main():
     if not opts.input_files:
         Emit.error("missing input file")
 
-    parser = get_parser(opts)
     serializers = get_serializers(opts)
     patcher = get_patcher(opts)
 
     if not serializers:
         Emit.error("missing output directives")
 
+    supplementary_nodes = []
+    if opts.isar_includes:
+        if opts.isar:
+            Emit.error('Bad usage. Isar include is intended to supplement code '
+                       'of different language.\nPass it all together as input files.')
+        if not opts.sack:
+            Emit.error('Isar defines inclusion is supported only in "sack" compilation mode.')
+
+        isar_parser = get_isar_parser()
+
+        def supplementary_parser(*parse_args):
+            return parse_content(isar_parser, patcher, *parse_args)
+
+        supple_file_parser = FileProcessor(supplementary_parser, opts.include_dirs)
```

Comment:
This comment doesn't relate to PR, it's just something that bothered me about this piece of code right from start. You can safely ignore it.

This 3-part piece of code (written by me originally) seems indecipherable to me now. After giving it a thought, I realize that this actually constructs a 3-level hierarchy of parsers:
- definitions level (structs, enums, etc.)
- syntax tree level (references, sizes, alignments)
- file level (includes)

This would be much more to the point:
```python
isar_parser = get_isar_parser()
full_parser = FullParser(isar_parser, patcher)
file_parser = FileProcessor(full_parser, opts.include_dirs)
```
and FullParser would substitute two mysterious free-floating functions:
- ad-hoc functor using `parse_content` with undisclosed `*args` and two bound variables (God help us),
- helper `parse_content`, which does almost too much to be a simple helper hidden among `get_basename`-s,

```python
class FullParser():
    def __init__(self, parser, patcher):
        self.parser = parser
        self.patcher = patcher
    def __call__(self, *parse_args):
        nodes = self.parser.parse(*parse_args)
        if self.patcher:
            self.patcher(nodes)
        model.topological_sort(nodes)
        model.cross_reference(nodes, warn=Emit.warn)
        model.evaluate_kinds(nodes)
        model.evaluate_sizes(nodes, warn=Emit.warn)
        return nodes
```

Reply:
Brilliant. Applied.